org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -91,12 +96,19 @@ public class GradleProjectImporter extends AbstractProjectImporter {
 	public static final String BUILD_GRADLE_KTS_DESCRIPTOR = "build.gradle.kts";
 	public static final String SETTINGS_GRADLE_DESCRIPTOR = "settings.gradle";
 	public static final String SETTINGS_GRADLE_KTS_DESCRIPTOR = "settings.gradle.kts";
+	public static final String GRADLE_WRAPPER_PROPERTIES_DESCRIPTOR = "gradle/wrapper/gradle-wrapper.properties";
 
 	public static final GradleDistribution DEFAULT_DISTRIBUTION = GradleDistribution.forVersion(GradleVersion.current().getVersion());
 
 	public static final String IMPORTING_GRADLE_PROJECTS = "Importing Gradle project(s)";
 
 	public static final String COMPATIBILITY_MARKER_ID = IConstants.PLUGIN_ID + ".gradlecompatibilityerrormarker";
+	public static final String GRADLE_UPGRADE_WRAPPER_MARKER_ID = IConstants.PLUGIN_ID + ".gradleupgradewrappermarker";
+
+	public static final String GRADLE_INVALID_TYPE_CODE_MESSAGE = "Exact exceptions are not shown due to an outdated Gradle version, please consider to update your Gradle version.";
```

Comment:
I'd explicitly tell user what Gradle version is required.

For example, change the message to "Exact exceptions are not shown due to a lower Gradle version, please consider to update your Gradle version to 7.2 and above."